file_watcher/lastrun_file_monitor.py

```diff
@@ -0,0 +1,271 @@
+        __init__.
+        """
+        if (
+            self.latest_known_run_from_db is not None
+            and self.last_recorded_run_from_file is not None
+            and int(self.latest_known_run_from_db) < int(self.last_recorded_run_from_file)
+        ):
+            logger.info(
+                "Recovering lost runs between %s and%s", self.last_recorded_run_from_file, self.latest_known_run_from_db
+            )
+            if self.latest_known_run_from_db is not None and self.last_recorded_run_from_file is not None:
+                await self.recover_lost_runs(self.latest_known_run_from_db, self.last_recorded_run_from_file)
+            self.latest_known_run_from_db = self.last_recorded_run_from_file
+
+    def get_latest_run_from_db(self) -> Union[str, None]:
+        """
+        Retrieve the latest run from the database
+        :return: Return the latest run for the instrument that is set on this object
+        """
+        # This likely contains NDX<INSTNAME> so remove the NDX and go for it with the DB
+        actual_instrument = self.instrument[3:]
+        return self.db_updater.get_latest_run(actual_instrument)
+
+    async def watch_for_new_runs(self, run_once: bool = False) -> None:
+        """
+        This is the main loop for waiting for new runs and triggering
+        :param run_once: Defaults to False, and will only run the loop once, aimed at simplicity for testing.
+        """
+        logger.info("Starting watcher...")
+        run = True
+        while run:
+            if run_once:
+                run = False
+            current_time = datetime.datetime.now()
+            time_between_cycle_folder_checks = current_time - self.last_cycle_folder_check
+            # If it's been 6 hours do another check for the latest folder
+            if time_between_cycle_folder_checks.total_seconds() > 21600:
+                self.latest_cycle = self.get_latest_cycle()
+                self.last_cycle_folder_check = current_time
+
+            try:
+                run_in_file = self.get_last_run_from_file()
+            except RuntimeError as exception:
+                # Correctly handle problems with parsing the last run file by just trying again, this is likely an
+                # error in file integrity and should be fixed by the time we check again but happens often enough
+                # for this fix to be necessary
+                logger.exception(exception)
+                continue
+            if run_in_file != self.last_recorded_run_from_file:
+                logger.info("New run detected: %s", run_in_file)
+                # If difference > 1 then try to recover potentially missed runs:
+                if int(run_in_file) - int(self.last_recorded_run_from_file) > 1:
+                    await self.recover_lost_runs(self.last_recorded_run_from_file, run_in_file)
+                else:
+                    await self.new_run_detected(run_in_file)
```

Comment:
```suggestion
self._check_for_missed_runs(run_in_file)
```
Similar to above,  I think it would be beneficial to keep this most important loop as clean as possible

Reply:
Another good idea